src/draive/openai/chat.py

```diff
@@ -201,3 +363,54 @@ async def _chat_tool_call(
             "tool_call_id": call_id,
             "content": "Error",  # TODO: refine error result message
         }
+
+
+async def _complete_chat_tool_calls(  # noqa: C901, PLR0912
+    *,
+    tool_calls: list[ChoiceDeltaToolCall],
+    completion_stream: AsyncStream[ChatCompletionChunk],
+) -> list[ChatCompletionMessageToolCall]:
+    # iterate to the over the stream to get full list of tool calls
+    async for chunk in completion_stream:
+        for call in chunk.choices[0].delta.tool_calls or []:
+            try:
+                tool_call: ChoiceDeltaToolCall = next(
+                    tool_call for tool_call in tool_calls if tool_call.index == call.index
+                )
+
+                if call.id:
+                    if tool_call.id is not None:
+                        tool_call.id += call.id
+                    else:
+                        tool_call.id = call.id
+                else:
+                    pass
+
+                if call.function is None:
+                    continue
+
+                if tool_call.function is None:
+                    tool_call.function = call.function
+                    continue
+
+                if call.function.name:
+                    if tool_call.function.name is not None:
+                        tool_call.function.name += call.function.name
+                    else:
+                        tool_call.function = call.function
+                else:
+                    pass
+
+                if call.function.arguments:
+                    if tool_call.function.arguments is not None:
+                        tool_call.function.arguments += call.function.arguments
+                    else:
+                        tool_call.function = call.function
```

Comment:
What is the case when interpreter hits this line? We already handled empty and None values.

Reply:
copy/paste error - it should set the name not the whole function - when function calling is requested names of functions may not be in place yet - fixed